client/transport/tcp.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+
+module.exports = options => {
```

Comment:
This file contains duplicate code similar to http.js

Reply:
Try to move this code to common abstraction layer

Reply:
I will try)